src/components/prompt_box.tsx

```diff
@@ -0,0 +1,53 @@
+import { title } from 'process';
+import { SetStateAction, useState } from 'react';
+import { Button } from 'react-bootstrap';
+import Modal from 'react-bootstrap/Modal';
+
+interface InputModalProps {
+    isOpen: boolean;
```

Comment:
We should probably also pass is a validateInput method and show an error if it is invalid.

For a job name, the validation should:
- Restrict the character set to [a-zA-Z0-9\-_#:>]; are there other characters we should allow? 
- Not allow a duplicate with an existing job name.

Maybe this should be an `inputValidators` array of {validator, errorMsg} objects instead so we can provide different error messages depending on which test failed.